src/app/inventory/ItemIcon.tsx

```diff
@@ -0,0 +1,154 @@
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
+import BungieImage, { bungieNetPath } from 'app/dim-ui/BungieImage';
+import {
+  DestinyEnergyTypeDefinition,
+  DestinyInventoryItemDefinition,
+} from 'bungie-api-ts/destiny2';
+import clsx from 'clsx';
+import { BucketHashes, ItemCategoryHashes } from 'data/d2/generated-enums';
+import React from 'react';
+import { DimItem } from './item-types';
+import styles from './ItemIcon.m.scss';
+
+const itemTierStyles = {
+  Legendary: styles.legendary,
+  Exotic: styles.exotic,
+  Common: styles.basic,
+  Rare: styles.rare,
+  Uncommon: styles.common,
+};
+
+/**
+ * This is just the icon part of the inventory tile - without the bottom stats bar, tag icons, etc.
+ * This exists because we have to do a fair bit of work to make the icon look like it does in game
+ * with respect to masterwork, season icons, mod overlays, etc.
+ *
+ * This renders just a fragment - it always needs to be rendered inside another div with class "item".
+ */
+export default function ItemIcon({ item, className }: { item: DimItem; className?: string }) {
+  const isCapped = item.maxStackSize > 1 && item.amount === item.maxStackSize && item.uniqueStack;
+  const borderless =
+    (item?.destinyVersion === 2 &&
+      (item.bucket.hash === BucketHashes.Subclass ||
+        item.itemCategoryHashes.includes(ItemCategoryHashes.Packages))) ||
+    item.isEngram;
+  const itemImageStyles = clsx('item-img', className, {
+    [styles.complete]: item.complete || isCapped,
+    [styles.borderless]: borderless,
+    [styles.masterwork]: item.masterwork,
+    [itemTierStyles[item.tier]]: !borderless && !item.plug,
+  });
+
+  return (
+    <>
+      <BungieImage src={item.icon} className={itemImageStyles} alt="" />
+      {item.masterwork && (
+        <div
+          className={clsx(styles.masterworkOverlay, { [styles.exoticMasterwork]: item.isExotic })}
+        />
+      )}
+      {item.iconOverlay && (
+        <div className={clsx(styles.iconOverlay)}>
+          <BungieImage src={item.iconOverlay} />
+        </div>
+      )}
+      {item.plug?.costElementIcon && (
+        <>
+          <div
+            style={{ backgroundImage: `url("${bungieNetPath(item.plug.costElementIcon)}")` }}
```

Comment:
for the second time this week i wonder "why isnt this a bungiebackgroundstyle"
and don't remember if i figured it out last time or just forgot

Reply:
It really should be, though I think I can redo it as just an img

Reply:
it doesn't need to be an image i think because there's only a few overlays and it doesn't need loading. i'd like to try compositing all overlays into a single multi-bg-image style once this is in